cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -625,11 +653,58 @@ async def _get_trustworthiness_score_async(
                     "log": response_json["log"],
                 }
 
-            return cast(float, response_json["confidence_score"])
+            return {"trustworthiness_score": response_json["confidence_score"]}
 
+        except RateLimitError as e:
+            if capture_exceptions:
+                warning_message = (
+                    f"prompt[{batch_index}] failed. Worth retrying. Error: {str(e.message)}"
+                )
+                warnings.warn(warning_message)
+
+                return TLMScore(
+                    trustworthiness_score=None,
+                    log={"error": {"message": str(e.message), "retryable": True}},
+                )
+            raise e
+        except TlmBadRequest as e:
+            if capture_exceptions:
+                retry_message = (
+                    "Worth retrying."
+                    if e.retryable
+                    else "Retrying will not help. Please address the issue described in the error message before attempting again."
+                )
+                warning_message = (
+                    f"prompt[{batch_index}] failed. {retry_message} Error: {str(e.message)}"
+                )
+                warnings.warn(warning_message)
+
+                return TLMScore(
+                    trustworthiness_score=None,
+                    log={"error": {"message": str(e.message), "retryable": e.retryable}},
+                )
+            raise e
+        except TlmServerError as e:
+            if capture_exceptions:
+                warning_message = (
+                    f"prompt[{batch_index}] failed. Worth retrying. Error: {str(e.message)}"
+                )
+                warnings.warn(warning_message)
+
+                return TLMScore(
+                    trustworthiness_score=None,
+                    log={"error": {"message": str(e.message), "retryable": True}},
+                )
+            raise e
```

Comment:
This seems to be exactly duplicated with the `_prompt_async()` method, maybe consider putting this in some helper function?